### ISSUE TITLE
Diagnostics in umbrella apps eex files have the wrong path

### ISSUE THREAD

**Issue by axelson** [elixir-lsp/elixir-ls#247]
Diagnostics returned for eex files in an umbrella app have the wrong path

For a line with a compilation error: `<% IO.puts() %>` in `/home/jason/dev/tmp/hello_phoenix_umbrella/apps/hello_phoenix_web/lib/hello_phoenix_web/templates/page/index.html.eex`

This diagnostic is returned:
```
{
	"resource": "/home/jason/dev/tmp/hello_phoenix_umbrella/lib/hello_phoenix_web/templates/page/index.html.eex",
	"owner": "_generated_diagnostic_collection_name_#0",
	"severity": 4,
	"message": "IO.puts/0 is undefined or private. Did you mean one of:\n\n      * puts/1\n      * puts/2",
	"source": "Elixir",
	"startLineNumber": 4,
	"startColumn": 1,
	"endLineNumber": 4,
	"endColumn": 1
}
```

Instead the resource path should be `/home/jason/dev/tmp/hello_phoenix_umbrella/apps/hello_phoenix_web/lib/hello_phoenix_web/templates/page/index.html.eex`

Note: the Elixir compiler may be returning just `lib/hello_phoenix_web/templates/page/index.html.eex`

In my testing this only applies to EEx templates, diagnostics in normal .ex files appears to be fine.

Related: #241

**Comment by lukaszsamson** [comment]
Elixir compiler is returning relative path here
```
%Mix.Task.Compiler.Diagnostic{
    file: "lib/phx_error_path_umbr_web/templates/page/index.html.heex",
    severity: :warning,
    message: ["IO.puts/0", " is undefined or private",
     ". Did you mean:\n\n      * puts/1\n      * puts/2\n"],
    position: 5,
    compiler_name: "Elixir",
    details: nil
  }
```
In contrast to other cases when it returns absolute path
```
%Mix.Task.Compiler.Diagnostic{
    file: "/Users/lukaszsamson/bug_repro/phx_error_path_umbr_umbrella/apps/phx_error_path_umbr_web/lib/phx_error_path_umbr_web/views/page_view.ex",
    severity: :error,
    message: "** (EEx.SyntaxError) lib/phx_error_path_umbr_web/templates/page/some.html.eex:2:1: missing token '%>'\n    (eex 1.14.0-dev) lib/eex.ex:224: EEx.compile_string/2\n    (phoenix_view 1.1.2) lib/phoenix/template.ex:419: Phoenix.Template.compile/3\n    (phoenix_view 1.1.2) lib/phoenix/template.ex:197: anonymous fn/4 in Phoenix.Template.\"MACRO-__before_compile__\"/2\n    (elixir 1.14.0-dev) lib/enum.ex:2462: Enum.\"-reduce/3-lists^foldl/2-0-\"/3\n    (phoenix_view 1.1.2) expanding macro: Phoenix.Template.__before_compile__/1\n    lib/phx_error_path_umbr_web/views/page_view.ex:1: PhxErrorPathUmbrWeb.PageView (module)\n",
    position: 2,
    compiler_name: "Elixir",
    details: nil
  }
```
```
%Mix.Task.Compiler.Diagnostic{
    file: "/Users/lukaszsamson/bug_repro/phx_error_path_umbr_umbrella/apps/phx_error_path_umbr_web/lib/phx_error_path_umbr_web/views/page_view.ex",
    severity: :warning,
    message: ["IO.trtttrr/0", " is undefined or private", ""],
    position: 3,
    compiler_name: "Elixir",
    details: nil
  }
```
```
%Mix.Task.Compiler.Diagnostic{
    file: "/Users/lukaszsamson/bug_repro/phx_error_path_umbr_umbrella/apps/phx_error_path_umbr_web/lib/phx_error_path_umbr_web/views/page_view.ex",
    severity: :error,
    message: "** (SyntaxError) lib/phx_error_path_umbr_web/views/page_view.ex:3:26: unexpected token: ). The \"do\" at line 1 is missing terminator \"end\"\n    (elixir 1.14.0-dev) lib/kernel/parallel_compiler.ex:346: anonymous fn/5 in Kernel.ParallelCompiler.spawn_workers/7\n",
    position: 3,
    compiler_name: "Elixir",
    details: nil
  }
```

**Comment by lukaszsamson** [comment]
Fixed upstream on elixir 1.14+
